decompiler/frontend/binaryninja/handlers/constants.py

```diff
@@ -66,6 +75,21 @@ def _lift_symbol_pointer(self, address: int, symbol: bSymbol) -> Optional[Symbol
         if symbol.type in (SymbolType.ImportedFunctionSymbol, SymbolType.ExternalSymbol):
             return ImportedFunctionSymbol(symbol.name, address, vartype=Pointer(Integer.char()))
 
+    def _lift_import_address_symbol(self, bv: BinaryView, symbol: bSymbol):
+        """Lift entry from .got section: addresses(offsets) of external symbols:
+           - first lift the global variable pointed by the symbol.
+           - second construct &global_variable.
+        Lift IAT symbol:
+           - construct ImportedFunctionSymbol with the same name.
+        """
+        pointer_value = bv.read_pointer(symbol.address)
+        global_data_pointed_by_symbol = bv.get_data_var_at(pointer_value)
+        if global_data_pointed_by_symbol:
+            lifted_global_var = self._lifter.lift(global_data_pointed_by_symbol, bv=bv, parent_addr=symbol.address)
+            return UnaryOperation(OperationType.address, [lifted_global_var])
```

Comment:
It covers the case when there is an entry in .got table which is a _pointer_ on the entry with the same name in data section.

Reply:
I could move this functionality to `_lift_symbol_pointer`.

We need to differentiate 2 cases here since `ImportAddresSymbol` is being used by Binary Ninja for 2 cases:
- IAT entries, which should be lifted by their name, `ImportedFunctionSymbol(symbol.name...)`, exactly as you stated;
- .got entries, more precisely addresses of extern global variables.

Reply:
I see. I guess in this case it would be fine to extend `_lift_symbol_pointer` and reflect on why we use the address in some cases.

Reply:
Moved and added comment.